### PR TITLE
version of ALLEGRO with ECAL barrel with cell corners projective in phi

### PR DESCRIPTION
**Pull request by giovannimarchiori**
BEGINRELEASENOTES
- New version of ALLEGRO detector o1_v03 with ecal barrel with 11 layers with cell corners projective along phi. No changes to the other sub detectors.
ENDRELEASENOTES

### CODE REVIEW

**Comment by giovannimarchiori** [comment]
Hi @BrieucF thanks for having looked at the PR and spotted the little typo in the documentation. Is there anything else needed before merging?

Thanks, Giovanni

**Comment by BrieucF** [comment]
Hi Giovanni, yes sorry, I started to review then had to do something else. Can you add a test for this new ALLEGRO version? https://github.com/key4hep/k4geo/blob/main/lcgeoTests/CMakeLists.txt#L90-L94

**Comment by giovannimarchiori** [comment]
> Hi Giovanni, yes sorry, I started to review then had to do something else. Can you add a test for this new ALLEGRO version? https://github.com/key4hep/k4geo/blob/main/lcgeoTests/CMakeLists.txt#L90-L94

done

**Comment by giovannimarchiori** [comment]
downstream-build errors are unrelated

The following tests FAILED:
	  2 - t_test_CLIC_o3_v14_reco_conformal (SEGFAULT)
	  3 - t_test_CLIC_o3_v14_vtx_makentp (Failed)
	  5 - t_test_FCCee_o1_v04_reco_conformal (Subprocess aborted)

**Comment by giovannimarchiori** [comment]
I just added some comments to the C++ code that builds the LAr detector to make it a bit easier to follow the steps of the G4 construction, and some comments to the corresponding xml file to clarify the meaning of the vector with the layer thicknesses

**Comment by BrieucF** [comment]
Can you also use the v03 constructor for upstream and calibration xml's?

**Comment by giovannimarchiori** [comment]
Yes, when we have agreed on the rounding ;)

**Comment by BrieucF** [comment]
Ha sorry, missed the last mail, taking a look

**Comment by giovannimarchiori** [comment]
@BrieucF : I updated also the other xml files for v03. In the xml of v02 I clarified that the dimensions are electrode lengths scaled by calo dR/L(plane). In the cpp of v03 I also added a check that the electrode plane length in the xml is consistent within 0.5 mm with that calculated from calo R1, R2 and alpha

Let me check with Nicolas if he's OK with these non rounded numbers and then I think we're good to merge

**Comment by BrieucF** [comment]
Thanks Giovanni! LGTM!

File: detector/calorimeter/ECalBarrel_NobleLiquid_InclinedTrapezoids_o1_v03_geo.cpp [hunk 1]
```diff
@@ -0,0 +1,679 @@
...
+  double layersTotalHeight = 0;
+  for (dd4hep::xml::Collection_t layer_coll(layers, _Unicode(layer)); layer_coll; ++layer_coll) {
+    dd4hep::xml::Component layer = layer_coll;
+    numLayers += layer.repeat();
+    for (int iLay = 0; iLay < layer.repeat(); iLay++) {
+      layerHeight.push_back(layer.thickness());
+    }
+    layersTotalHeight += layer.repeat() * layer.thickness();
+  }
+  lLog << MSG::DEBUG << "Total electrode length from calorimeter xml description (cm): " << layersTotalHeight << endmsg;
```
Comment (andresailer) [inline detector/calorimeter/ECalBarrel_NobleLiquid_InclinedTrapezoids_o1_v03_geo.cpp:75]:
```suggestion
  lLog << MSG::DEBUG << "Total electrode length from calorimeter xml description (cm): " << layersTotalHeight/dd4hep::cm << endmsg;
```

File: detector/calorimeter/ECalBarrel_NobleLiquid_InclinedTrapezoids_o1_v03_geo.cpp [hunk 2]
```diff
@@ -0,0 +1,679 @@
...
+    double Lmax = runningHeight + layerHeight[iLay];
+    runningHeight += layerHeight[iLay];
+    double rMin = sqrt(Rmin*Rmin + Lmin*Lmin + 2*Rmin*Lmin*cos(angle));
+    double rMax = sqrt(Rmin*Rmin + Lmax*Lmax + 2*Rmin*Lmax*cos(angle));
+    lLog << MSG::INFO << "      layer " << iLay << " (cm) = " << rMin << " - " << rMax << endmsg;
+  }
+
+  // check that electrode length is consistent with calorimeter radial extent
+  // and inclination angle to within 0.5 mm
+  if (fabs(planeLength - layersTotalHeight) > 0.05) {
```
Comment (andresailer) [inline detector/calorimeter/ECalBarrel_NobleLiquid_InclinedTrapezoids_o1_v03_geo.cpp:295]:
```suggestion
  if (fabs(planeLength - layersTotalHeight) > 0.05*dd4hep::cm) {
```

**Review by andresailer — COMMENTED** [review]
One should also use units in the CPP implementation

File: detector/calorimeter/ECalBarrel_NobleLiquid_InclinedTrapezoids_o1_v03_geo.cpp (hunk 1 above)
Comment (giovannimarchiori) [reply detector/calorimeter/ECalBarrel_NobleLiquid_InclinedTrapezoids_o1_v03_geo.cpp]: Fixed in latest commit

**Comment by giovannimarchiori** [comment]
Fixed in latest commit

**Comment by giovannimarchiori** [comment]
Hello @andresailer, 
please let me know if I need to do further modifications or if the code can be merged as it is now.
Cheers,
  Giovanni

**Review by andresailer — APPROVED** [review]
Nothing else from my side.
Thanks!

**Comment by BrieucF** [comment]
Thanks Giovanni and Andre!

**Pull request merged**